### DISCUSSION TITLE
Paint it ⬛

### DISCUSSION THREAD

**Discussion by boisgera** [boisgera/python-fr#10] [General]
[Black](https://github.com/psf/black) est un formatteur de texte Python qui va vous aidez à adopter les conventions d'écriture "standard" en Python en faisant passer votre code dans une "moulinette" pour le normaliser. Par exemple, Black opérera les transformations suivantes :

  - `range (10)` -> `range(10)` (pas d'espace entre une fonction et ses arguments)
  
  - `a=1` -> `a = 1` (espace avant et après un `=` dans une assignation)

  - `return(True)` -> `return True` (`return` est un mot-clé, pas une fonction, les parenthèses ne sont pas nécessaires).

**TODO:**

  - Est-ce que Black est disponible [comme package conda](https://anaconda.org) ? Sur [PyPi](https://pypi.org/) (et donc installable avec pip)? 

  - Installez Black dans votre environnement conda dédié au jeu du serpent,

  - Utilisez-le sur vos fichiers Python et puis mettez à jour votre dépôt GitHub avec la nouvelle version de ces fichiers,

  - Examinez vos fichiers Python avant et après formattage par Black. Puis essayez de répérer en quoi votre style d'écriture diffère des conventions standard.

  - Optionnel : Il existe un plugin Black pour VS Code qui intègre ce formatteur à l'IDE. Vous pouvez l'installer puis essayer d'utiliser le formattage par Black depuis VS Code.

**Comment by GNRNicolas** [comment]
Comment installer Black? Je n'ai pas compris comment s'y prendre.

**Reply by boisgera** [reply]
Est-ce que tu as pris connaissance de la section "installation" ici https://pypi.org/project/black/ ?

**Comment by Jean-Sidati** [comment]
De mon côté j'ai bien installer black mais je ne parviens pas a l'utiliser (mon terminal me donne plein d'erreur de commande non reconnu lorsque j'essaie de reproduire ce qu'il y a ici https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file ).

**Reply by boisgera** [reply]
Tu peux essayer black sans le fichier de configuration (qui ne sert que pour un usage avancé)

**Reply by boisgera** [reply]
Tu peux aussi nous donner ton message d'erreur si ça ne marche toujours pas

**Reply by Jean-Sidati** [reply]
J'ai réussi a reformater en ignorant le fichier de configuration, merci beaucoup

**Reply by boisgera** [reply]
👍
